AGASC supplement processing failing

The current processing seems to be failing.
```
WARNING - 'agasc-update-magnitudes --log-level debug --output-dir /proj/sot/ska3/flight/data/agasc/rc --report' returned non-zero status: 256
2023-04-16 18:30:28,496   identifying outlying observations (OBSID=17898, mp_starcat_time=2016:088:21:54:47.566)
2023-04-16 18:30:28,496   identifying outlying observations (OBSID=18808, mp_starcat_time=2016:091:10:07:50.653)
2023-04-16 18:30:28,497   identifying outlying observations (OBSID=17899, mp_starcat_time=2016:263:06:32:39.175)
2023-04-16 18:30:28,497   identifying outlying observations (OBSID=19882, mp_starcat_time=2016:267:11:11:40.571)
2023-04-16 18:30:28,498   identifying outlying observations (OBSID=50688, mp_starcat_time=2016:267:23:55:18.419)
2023-04-16 18:30:28,498   identifying outlying observations (OBSID=20036, mp_starcat_time=2017:072:22:54:05.330)
2023-04-16 18:30:28,499   identifying outlying observations (OBSID=19290, mp_starcat_time=2017:075:04:07:33.998)
2023-04-16 18:30:28,499   identifying outlying observations (OBSID=19289, mp_starcat_time=2017:266:15:19:04.302)
2023-04-16 18:30:28,500   identifying outlying observations (OBSID=20927, mp_starcat_time=2018:072:23:50:12.293)
2023-04-16 18:30:28,500   identifying outlying observations (OBSID=21037, mp_starcat_time=2018:074:09:52:55.134)
2023-04-16 18:30:28,500   identifying outlying observations (OBSID=21038, mp_starcat_time=2018:077:23:34:33.598)
2023-04-16 18:30:28,501   identifying outlying observations (OBSID=20322, mp_starcat_time=2018:078:21:24:04.027)
2023-04-16 18:30:28,501   identifying outlying observations (OBSID=21042, mp_starcat_time=2018:082:08:20:45.650)
2023-04-16 18:30:28,502   identifying outlying observations (OBSID=21043, mp_starcat_time=2018:083:23:57:28.826)
2023-04-16 18:30:28,502   identifying outlying observations (OBSID=21044, mp_starcat_time=2018:085:05:10:15.300)
2023-04-16 18:30:28,503   identifying outlying observations (OBSID=21303, mp_starcat_time=2019:078:09:38:22.468)
2023-04-16 18:30:28,503   identifying outlying observations (OBSID=22155, mp_starcat_time=2019:082:14:26:23.212)
2023-04-16 18:30:28,503   identifying outlying observations (OBSID=22156, mp_starcat_time=2019:083:06:52:17.168)
2023-04-16 18:30:28,504   identifying outlying observations (OBSID=22424, mp_starcat_time=2020:083:20:19:57.416)
2023-04-16 18:30:28,504   identifying outlying observations (OBSID=23198, mp_starcat_time=2020:088:12:38:04.723)
2023-04-16 18:30:28,505   identifying outlying observations (OBSID=22425, mp_starcat_time=2020:256:19:09:07.110)
2023-04-16 18:30:28,505   identifying outlying observations (OBSID=24652, mp_starcat_time=2020:261:02:05:00.282)
2023-04-16 18:30:28,506   identifying outlying observations (OBSID=23533, mp_starcat_time=2021:084:18:14:42.948)
2023-04-16 18:30:28,506   identifying outlying observations (OBSID=22703, mp_starcat_time=2021:090:00:23:53.876)
2023-04-16 18:30:28,506   identifying outlying observations (OBSID=23534, mp_starcat_time=2021:298:20:16:23.403)
2023-04-16 18:30:28,507   identifying outlying observations (OBSID=24295, mp_starcat_time=2021:300:15:22:55.780)
2023-04-16 18:30:28,507   identifying outlying observations (OBSID=24654, mp_starcat_time=2021:301:03:51:55.265)
2023-04-16 18:30:28,508   identifying outlying observations (OBSID=27443, mp_starcat_time=2022:263:16:59:47.126)
2023-04-16 18:30:28,508   identifying outlying observations (OBSID=25514, mp_starcat_time=2022:264:03:11:27.974)
2023-04-16 18:30:28,508   identifying outlying observations (OBSID=25906, mp_starcat_time=2022:270:13:37:13.779)
2023-04-16 18:30:28,509   identifying outlying observations (OBSID=27605, mp_starcat_time=2022:351:20:02:02.217)
2023-04-16 18:30:28,509   identifying outlying observations (OBSID=27606, mp_starcat_time=2022:352:02:51:41.560)
2023-04-16 18:30:28,510   identifying outlying observations (OBSID=25511, mp_starcat_time=2023:103:19:12:20.573)
2023-04-16 18:30:32,108   stats for AGASC ID 1201542672:  8.178614346046494

progress: 100%|██████████| 228/228 [38:47<00:00, 10.21s/star]
2023-04-16 18:30:32,109 --------------------------------------------------------------------------------
2023-04-16 18:30:32,730 Got:
  2952 OBSIDs,  228 stars,  2 failed observations,
2023-04-16 18:30:32,731 Updating /proj/sot/ska3/flight/data/agasc/rc/mag_stats_agasc.fits
Traceback (most recent call last):
  File "/proj/sot/ska3/flight/bin/agasc-update-magnitudes", line 33, in <module>
    sys.exit(load_entry_point('agasc==4.14.1', 'console_scripts', 'agasc-update-magnitudes')())
  File "/proj/sot/ska3/flight/lib/python3.10/site-packages/agasc/scripts/update_mag_supplement.py", line 158, in main
    update_mag_supplement.do(
  File "/proj/sot/ska3/flight/lib/python3.10/site-packages/agasc/supplement/magnitudes/update_mag_supplement.py", line 572, in do
    update_mag_stats(obs_stats, agasc_stats, fails, output_dir)
  File "/proj/sot/ska3/flight/lib/python3.10/site-packages/agasc/supplement/magnitudes/update_mag_supplement.py", line 207, in update_mag_stats
    agasc_stats = _update_table(table.Table.read(filename), agasc_stats,
  File "/proj/sot/ska3/flight/lib/python3.10/site-packages/agasc/supplement/magnitudes/update_mag_supplement.py", line 176, in _update_table
    assert table_old.as_array().dtype.names == table_new.as_array().dtype.names, \
AssertionError: Tables have different dtype
```

--- a/agasc/supplement/magnitudes/mag_estimate.py
+++ b/agasc/supplement/magnitudes/mag_estimate.py
@@ -1064,7 +1064,12 @@ def get_agasc_id_stats(agasc_id, obs_status_override=None, tstop=None):
         'selected_atol': False,
         'selected_rtol': False,
         'selected_mag_aca_err': False,
-        'selected_color': False
+        'selected_color': False,
+        'f_mag_est_ok': 0,
+        'f_mag_est_ok_3': 0,
+        'f_mag_est_ok_5': 0,
+        'f_ok_3': 0,
+        'f_ok_5': 0,
     }
 
     for tag in ['dr3', 'dbox5']:

--- a/agasc/supplement/magnitudes/mag_estimate_report.py
+++ b/agasc/supplement/magnitudes/mag_estimate_report.py
@@ -285,14 +285,16 @@ class MagEstimateReport:
                     "ignore",
                     message="Warning: converting a masked element to nan.",
                 )
-                out.write(run_template.render(info=info,
-                                            sections=sections,
-                                            failures=fails,
-                                            star_reports=star_reports,
-                                            nav_links=nav_links,
-                                            tooltips=tooltips,
-                                            static_dir=static_dir,
-                                            glossary=GLOSSARY))
+                out.write(run_template.render(
+                    info=info,
+                    sections=sections,
+                    failures=fails,
+                    star_reports=star_reports,
+                    nav_links=nav_links,
+                    tooltips=tooltips,
+                    static_dir=static_dir,
+                    glossary=GLOSSARY
+                ))
 
         json_filename = filename.replace('.html', '.json')
         if json_filename == filename:

--- a/agasc/supplement/magnitudes/mag_estimate_report.py
+++ b/agasc/supplement/magnitudes/mag_estimate_report.py
@@ -243,7 +243,7 @@ class MagEstimateReport:
         star_reports = {}
         logger.debug('-' * 80)
         logger.info("Generating star reports")
-        for agasc_id in tqdm(np.atleast_1d(agasc_ids),
+        for agasc_id in tqdm(np.atleast_1d(agasc_ids).tolist(),
                              desc='progress', disable=no_progress, unit='star'):
             try:
                 logger.debug('-' * 80)

--- a/agasc/supplement/magnitudes/mag_estimate_report.py
+++ b/agasc/supplement/magnitudes/mag_estimate_report.py
@@ -5,6 +5,7 @@ import errno
 import os
 import copy
 import json
+import warnings
 from subprocess import Popen, PIPE
 from pathlib import Path
 from email.mime.text import MIMEText
@@ -140,10 +141,17 @@ class MagEstimateReport:
         plt.close(fig)
 
         with open(directory / 'index.html', 'w') as out:
-            out.write(star_template.render(agasc_stats=agasc_stat,
-                                           obs_stats=obs_stat.as_array(),
-                                           static_dir=static_dir,
-                                           glossary=GLOSSARY))
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore",
+                    message="Warning: converting a masked element to nan.",
+                )
+                out.write(star_template.render(
+                    agasc_stats=agasc_stat,
+                    obs_stats=obs_stat.as_array(),
+                    static_dir=static_dir,
+                    glossary=GLOSSARY)
+                )
         with open(directory / 'data.json', 'w') as json_out:
             json.dump(
                 {
@@ -272,14 +280,19 @@ class MagEstimateReport:
         if not self.directory.exists():
             self.directory.mkdir(parents=True)
         with open(self.directory / filename, 'w') as out:
-            out.write(run_template.render(info=info,
-                                          sections=sections,
-                                          failures=fails,
-                                          star_reports=star_reports,
-                                          nav_links=nav_links,
-                                          tooltips=tooltips,
-                                          static_dir=static_dir,
-                                          glossary=GLOSSARY))
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore",
+                    message="Warning: converting a masked element to nan.",
+                )
+                out.write(run_template.render(info=info,
+                                            sections=sections,
+                                            failures=fails,
+                                            star_reports=star_reports,
+                                            nav_links=nav_links,
+                                            tooltips=tooltips,
+                                            static_dir=static_dir,
+                                            glossary=GLOSSARY))
 
         json_filename = filename.replace('.html', '.json')
         if json_filename == filename:

--- a/agasc/supplement/magnitudes/update_mag_supplement.py
+++ b/agasc/supplement/magnitudes/update_mag_supplement.py
@@ -173,16 +173,21 @@ def get_agasc_id_stats_pool(agasc_ids, obs_status_override=None, batch_size=100,
 
 def _update_table(table_old, table_new, keys):
     # checking names, because actual types change upon saving in fits format
-    assert table_old.as_array().dtype.names == table_new.as_array().dtype.names, \
-        'Tables have different dtype'
+    if set(table_old.colnames) != set(table_new.colnames):
+        raise Exception(
+            'Tables have different columns:'
+            f'\n  {table_old.colnames}'
+            f'\n  {table_new.colnames}'
+        )
     table_old = table_old.copy()
     new_row = np.ones(len(table_new), dtype=bool)
     _, i_new, i_old = np.intersect1d(table_new[keys].as_array(),
                                      table_old[keys].as_array(),
                                      return_indices=True)
     new_row[i_new] = False
-    table_old[i_old] = table_new[i_new]
-    return table.vstack([table_old, table_new[new_row]])
+    columns = table_old.as_array().dtype.names
+    table_old[i_old] = table_new[i_new][columns]
+    return table.vstack([table_old, table_new[new_row][columns]])
 
 
 def update_mag_stats(obs_stats, agasc_stats, fails, outdir='.'):

--- a/agasc/supplement/magnitudes/update_mag_supplement.py
+++ b/agasc/supplement/magnitudes/update_mag_supplement.py
@@ -674,6 +674,13 @@ def do(start,
         except Exception as e:
             report_dir = output_dir
             logger.error(f'Error when creating report: {e}')
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            if exc_type is not None:
+                trace = traceback.format_exception(exc_type, exc_value, exc_traceback)
+                for level in trace:
+                    for line in level.splitlines():
+                        logger.debug(line)
+                    logger.debug('')
         finally:
             report_data_file = report_dir / report_data_file.name
             if not report_dir.exists():

